.github/workflows/tf-apply-hash.yml

```diff
@@ -0,0 +1,129 @@
+name: HASH Terraform Apply
+
+# TODO: While this workflow is specific to HASH, we should consider
+#  making this a reusable workflow, parameterized on the directory and environment.
+
+on:
+  workflow_dispatch:
+    inputs:
+      env:
+        type: choice
+        options: ["prod"]
+        description: "Environment to deploy"
+        required: true
+        default: "prod"
+
+  push:
+    branches:
+      - main
+    paths:
+      - "infra/internal-api/**"
+
+env:
+  VAULT_ADDR: ${{ secrets.VAULT_ADDR }}
+
+jobs:
+  matrix-eval:
+    name: Prepare environments to be applied
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - name: Set matrix
+        id: set-matrix
+        env:
+          ENV: ${{ github.event.inputs.env }}
+        run: |
+          # If no env is specified, run both dev and prod
+          if [ "$ENV" == "prod" ]; then
+            echo "matrix=[\"$ENV\"]" >> $GITHUB_OUTPUT
+          else
+            echo "matrix=[\"prod\"]" >> $GITHUB_OUTPUT
+          fi
+
+  terraform:
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      contents: read
+    env:
+      working-directory: infra/terraform/hash
+      TF_VAR_in_ci: true
```

Comment:
Is this only to work around parsing environment variables from Terraform by e.g. calling a bash script that outputs the environment as JSON?

Reply:
I mean the way how we pass in the `is_ci` environment variable. The `CI` variable is set by GH Actions automatically, I wonder if we want to rely on that instead of manually setting it.

Reply:
Yeah i was looking into that and having TF look for the `CI` env var, which may be possible. I don't think we can do `TF_VAR_in_ci: ${{ env.CI }}` as this is defined for the job and the `env` isn't set at this point as far as I can tell

Reply:
I suppose that could work, I just didn't want to introduce a `data` shell script like that. What if I look into doing this in the next-up asana task https://app.asana.com/0/1204610208832639/1204691860848733/f
and update
https://github.com/hashintel/hash/blob/70cd4979dbcae33947e7172f739a36641396cea5/infra/terraform/modules/variables/main.tf
to do that CI check?

Reply:
I'm fine with deferring it or not using it at all in this case, just wanted to check if we evaluated all possible solutions 🙂